### PR TITLE
Make SB release PR against main always

### PR DESCRIPTION
**Pull request by premun**
https://github.com/dotnet/installer/pull/15796

### CODE REVIEW

File: eng/templates/stages/release.yml [hunk 1]
```diff
@@ -367,7 +362,7 @@ stages:
             echo "  --sdkVersion $(sdkVersion)"
             echo "  --title $title"
             echo "  --body $body"
-            echo "  --targetBranch $target_branch"
+            echo "  --targetBranch main"
```
Comment (MichaelSimons) [inline eng/templates/stages/release.yml:365]: This doesn't look right.  "main" should only be used for preview releases.  The PRs for servicing releases (e.g. 6.0/7.0) should still me made against the servicing branches.

Comment (premun) [reply eng/templates/stages/release.yml:365]: Ah, you're right. Too of a hasty change. Will fix

**Review by mthalman — COMMENTED** [review]
The [submit script](https://github.com/dotnet/source-build/blob/main/eng/submit-source-build-release-pr.sh ) also needs to be updated to account for the changes in Versions.props from https://github.com/dotnet/installer/pull/15603. That would be specific to the target branch though.

**Comment by premun** [comment]
@mthalman thanks - I wanted to make that change separately as I need more time to read into it

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**